react/FilterNavigator.js

```diff
@@ -66,6 +72,94 @@ class FilterNavigator extends Component {
     hiddenFacets: {},
   }
 
+  state = {
+    openContent: false,
+    filtersChecks: {},
+    pagesArgs: {},
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const nonEmptyFilters = this.mapCheckboxFilters(this.filters.filter(spec => spec.options.length > 0))
+
+    if (this.willUpdateFilters(keys(nonEmptyFilters), keys(prevState.filtersChecks))) {
+      this.setState({ filtersChecks: nonEmptyFilters })
+    }
+  }
+
+  willUpdateFilters = (newFilters, prevFilters) => {
+    if (newFilters.length !== prevFilters.length) return true
+    return !newFilters.every(e => prevFilters.includes(e))
+  }
+
+  mapCheckboxFilters = (filters) => {
+    const { map, rest } = this.props
+
+    const mapperOption = (isSelected, selectedFilters) => opt => {
+      const pagesArgs = formatFacetToLinkPropsParam(opt.type, opt)
+      return {
+        name: opt.Name,
+        ...pagesArgs,
+        checked: isSelected || !!find(propEq('Name', opt.Name), selectedFilters),
+      }
+    }
+    const mapperOptionsForFilter = (map, rest) => ({ options, type }) => mountOptions(options, type, map, rest).map(mapperOption(false, this.selectedFilters))
```

Comment:
why do you need this wrapper function since it's only being used once and uses the same value of `map` and `rest` as on the upper scope?

it's generally more readable to place these mapper and filter functions inline, you should only extract them when you need reuse or they're too complex.